components/MapSection/MapSection.tsx

```diff
@@ -48,8 +48,9 @@ const renderLegendItems = (range?: string[], colours?: string[]): ReactNode => {
 
 const getComparePath = (router: NextRouter): string => {
   const pathname = router.asPath.split('?')[0].split('#')[0];
+  const queryString = router.asPath.split('?')[1];
 
-  return `${pathname}${pathname.endsWith('/') ? '' : '/'}compare`;
+  return `${pathname}${pathname.endsWith('/') ? '' : '/'}compare${queryString ? '?' + queryString : ''}`;
```

Comment:
We only need the locations ... you can get them from the router.query, via the LOCATION_CODE_QUERY & LOCATION_NAME_QUERY.  We don't need the other stuff.